src/harness/testcases/WINNF_FT_S_PCR_testcase.py

```diff
@@ -0,0 +1,671 @@
+      }
+    }
+    pal_record['userId'] = "pal_user"
+    pal_record['palId'] = "pal/10-2017/123456789/A"
+
+    # Step 2: Load the device_a in DP Test Harness
+    device_a = json.load(
+      open(os.path.join('testcases', 'testdata', 'device_a.json')))
+    device_a['installationParam']['latitude'] = 39.0119
+    device_a['installationParam']['longitude'] = -98.4842
+    device_a['userId'] = pal_record['licensee']
+
+    # Step 2: Load the device_b in DP Test Harness
+    device_b = json.load(
+      open(os.path.join('testcases', 'testdata', 'device_b.json')))
+    device_b['installationParam']['latitude'] = 39.139836
+    device_b['installationParam']['longitude'] = -97.770094
+    device_b['userId'] = pal_record['licensee']
+
+    url = 'sas_testUUT_url will be provided in the future'
+
+    config = {
+        'url': url,
+        'device_a': device_a,
+        'device_b': device_b,
+        'pal_record': pal_record
+    }
+    writeConfig(filename, config)
+
+  @configurable_testcase(generate_PCR_1_default_config)
+  def test_WINNF_FT_S_PCR_1(self, config_filename):
+    # Load the Config file
+    config = loadConfig(config_filename)
+    url = config['url']
+    device_a = config['device_a']
+    pal_record = config['pal_record']
+    device_b = config['device_b']
+
+    # Step 1: Inject PAL record
+    self._sas_admin.InjectPalDatabaseRecord(pal_record)
+    # Step 2: register device_a in SAS UUT
+    self._sas_admin.InjectFccId({'fccId': device_a['fccId']})
+    self._sas_admin.InjectUserId({'userId': device_a['userId']})
+    request = {'registrationRequest': [device_a]}
+    response = self._sas.Registration(request)['registrationResponse'][0]
+    # Check registration response of device_a in SAS UUT
+    self.assertEqual(response['response']['responseCode'], 0)
+    del request, response
+    # Step 2: register device_b in SAS UUT
+    self._sas_admin.InjectFccId({'fccId': device_b['fccId']})
+    self._sas_admin.InjectUserId({'userId': device_b['userId']})
+    request = {'registrationRequest': [device_b]}
+    response = self._sas.Registration(request)['registrationResponse'][0]
+    # Check registration response of device_b in SAS UUT
+    self.assertEqual(response['response']['responseCode'], 0)
+    del request, response
+    # Step 3: Triggers SAS UUT to create a PPA boundary
+    cbsdIds = [device_a['cbsdSerialNumber'],device_b['cbsdSerialNumber']]
+    palIds = [pal_record['palId']]
+    response = self._sas_admin.TriggerPpaCreation(cbsdIds, palIds)
+    self.assertEqual(response['response']['responseCode'], 0)
+    self.assertNotEqual(response['ppaId'], None)
+    # Step 4: Triggers the SAS UUT to generate the Full Activity Dump
+    self._sas_admin.TriggerCreateActivityDump()
+    # Step 5: SAS Test Harness uses the SAS-SAS protocol to retrieve the PPA zone
+
+    full_activity_dump = _RequestGet('https://%s' %
+                (url + '/dump'))
+    self.polygon_dump = _RequestGet('https://%s' %
```

Comment:
(1) This will not retrieve the data in the format that is expected later.
(2) This assumes that the URL provided by the SAS UUT is not prefixed with "https", which cannot be assumed.
(3) We should not be setting member variables in a test case.

Reply:
In the function descriptions, it seems return JSON response received from server. I think it could return the ppa dump object? Is there any other ways to get the ppa object from SAS UUT?

Reply:
Here I noticed there is an example of using _RequestGet() it has 'https' as a part of url?

 def _SasRequest(self, method_name, request, ssl_cert=None, ssl_key=None):
    return _RequestGet('https://%s/%s/%s/%s' %
                       (self._base_url, self._sas_version, method_name, request),
                       self._tls_config.WithClientCertificate(
                           ssl_cert or self._GetDefaultSasSSLCertPath(),
                           ssl_key or self._GetDefaultSasSSLKeyPath()))

Reply:
Added   the new method GetSasDumpRecord and      GetSasPPADumpRecord in sas.py to pull FAD records from SAS UTT and fetch   PPA zone from dump record to     calculate maximum ppa boundary